### PR TITLE
Introducing Log-state-enum for new schema LogList

### PR DESCRIPTION
**Pull request by Mercurrent**
To be able to operate on those states and compare them.

### CODE REVIEW

File: loglist2/loglist2.go [hunk 1]
```diff
@@ -136,28 +150,38 @@ type TreeHead struct {
 	TreeSize int64 `json:"tree_size"`
 }
 
-// String method returns printable name of the state.
-func (ls *LogStates) String() string {
+// Which method returns Log-status for descriptive struct.
```
Comment (gdbelvin) [inline loglist2/loglist2.go]: `Which` is a little generic. How about `LogStatus` ?

File: loglist2/loglist2.go [hunk 2]
```diff
@@ -136,28 +150,38 @@ type TreeHead struct {
...
-		return "Rejected"
+		return RejectedLogStatus
 	default:
-		return "Empty"
+		return UndefinedLogStatus
 	}
 }
 
+// String returns printable name for log-state.
+func (ls LogStatus) String() string {
```
Comment (gdbelvin) [inline loglist2/loglist2.go]: Group this up with the definition of `LogStatus`? 

Also, this looks like the *perfect* use case for https://godoc.org/golang.org/x/tools/cmd/stringer

File: loglist2/loglist2.go [hunk 3]
```diff
@@ -97,6 +97,20 @@ type TemporalInterval struct {
 	EndExclusive time.Time `json:"end_exclusive"`
 }
 
+// LogStatus indicates Log status.
+type LogStatus int
+
+// LogStatus values
+const (
```
Comment (gdbelvin) [inline loglist2/loglist2.go]: 👍

File: loglist2/loglist2.go (hunk 2 above)
Comment (Mercurrent) [reply loglist2/loglist2.go]: Removed. As stringer puts its production in a separate file.

File: loglist2/loglist2.go (hunk 1 above)
Comment (Mercurrent) [reply loglist2/loglist2.go]: Renamed.

**Pull request merged**